src/main/kotlin/snyk/common/lsp/Types.kt

```diff
@@ -236,6 +489,31 @@ data class IssueData(
     }
 
     override fun hashCode(): Int {
+        if (this.getProductType() == ProductType.OSS) {
+            var result = ruleId.hashCode()
+            result = 31 + description.hashCode()
```

Comment:
I'm not sure if this is the reason but I get this error when I run the plugin:

```
java.lang.NullPointerException: Cannot invoke "String.hashCode()" because "this.ruleId" is null
	at snyk.common.lsp.IssueData.hashCode(Types.kt:493)
	at snyk.common.lsp.ScanIssue.hashCode(Types.kt)
	at java.base/java.util.HashMap.hash(HashMap.java:338)
	at java.base/java.util.HashMap.put(HashMap.java:610)
	at java.base/java.util.HashSet.add(HashSet.java:221)
```
and the difference I see it's the calculation for the hash for `description` is using `+` instead of `*` operator .

Reply:
I think we are missing the `* result`

Reply:
Already discussed this in Slack but this runtime error is caused because https://github.com/snyk/snyk-ls/releases/tag/v20240513.094925 is required :)